### PR TITLE
UX like OSX with snap points

### PR DESCRIPTION
**Pull request by gastonmorixe**
UX 2 like OSX

### CODE REVIEW

File: demo/Examples.js [hunk 1]
```diff
@@ -24,6 +24,121 @@ const BasicVerticalExample = () => {
...
+            primary="second"
+            defaultSize={max}
+            maxSize={max}
+            minSize={min}
+            style={{ backgroundColor: 'red', position: 'static' }}
+            pane1Style={{ backgroundColor: 'pink' }}
+            pane2Style={{ backgroundColor: 'yellow' }}
+            // paneStyle={{ transition: 'all .1s ease' }}
+            controlSnap={({jumped, newPosition, current, draggingDelta, startPosition, setSize, setSizeOnRelease, setStep}) => {
+                console.log("PANE 2", {current, draggingDelta})
```
Comment (marcinbunsch) [inline demo/Examples.js]: Remove stray console.log

File: demo/Examples.js [hunk 2]
```diff
@@ -24,6 +24,121 @@ const BasicVerticalExample = () => {
...
+    return (
+        <SplitPane 
+            split="vertical"
+            style={{ position: 'static' }}
+            defaultSize={pointA}
+            maxSize={max}
+            minSize={min}
+            // paneStyle={{ transition: 'all .1s ease' }}
+            controlSnap={({jumped, current, newPosition, draggingDelta, startPosition, setSize, setSizeOnRelease, setStep}) => {
+                console.log("PANE 1", {jumped, newPosition, current})
```
Comment (marcinbunsch) [inline demo/Examples.js]: Remove stray console.log

**Comment by marcinbunsch** [comment]
Few small things:
  - stray console log
  - inconsistent formatting
  - commented out code

Additionally, as we're pulling in this lib in our own fork, we own it, meaning we need to have at least two people familiarized with this lib capable of doing fixes should need be.

File: package.json [hunk 3]
```diff
@@ -3,7 +3,7 @@
   "description": "React split-pane component",
   "main": "index.js",
   "types": "index.d.ts",
-  "version": "0.1.71",
+  "version": "0.1.4",
```
Comment (marcinbunsch) [inline package.json]: Why this change?

File: src/SplitPane.js [hunk 4]
```diff
@@ -22,174 +22,289 @@ function unFocus(document, window) {
...
+              setSizeOnRelease,
+              setStep,
+            });
+
+            this.state.draggingDelta = draggingDelta - positionDelta;
+            this.state.cleanDraggingDelta = this.state.draggingDelta;
+
+            if (controlSnapResult && controlSnapResult.constructor === Object) {
+              if (controlSnapResult.type === controlSnapTypes.STEP) {
+                // TODO
```
Comment (marcinbunsch) [inline src/SplitPane.js]: TODO?

**Review by marcinbunsch — CHANGES_REQUESTED** [review]

**Comment by gastonmorixe** [comment]
@marcinbunsch I agree with all you said.

File: package.json (hunk 3 above)
Comment (gastonmorixe) [reply package.json]: Well, the idea was to upgrade it to be sure npm takes it, sometimes it didn't. Still, I failed "upgrading it". I am not sure what version do you suggest to have here. same as original?

File: src/SplitPane.js (hunk 4 above)
Comment (gastonmorixe) [reply src/SplitPane.js]: This is for a future feature.

**Comment by gastonmorixe** [comment]
Formatting should be ok since I am running `npm run prettier` before committing.
